### PR TITLE
Make boards a partitioned-page component and add split routes

### PR DESCRIPTION
**Pull request by oliverguenther**
Adds a split view to the boards page:

- [x] Makes board view share the same template as the partitioned query space components (We cannot reuse the component completely due to the tight coupling to Query loading, which Boards do not have)
- [x] Refactors board filters into a service that was needed due breaking apart boards/board-list-container so they no longer are the same components (board-list-container is mounted in `content-left`, similar to the table)
- [x] Moved board routes into their own definition file

**Bugs**

- [x] When being on a free board, and I update the title, type or assignee in the split screen, the belonging card does not change.
- [x] When I have the split screen open, a click on the info icon of another card only selects this card. I would expect it, to show the details in the split screen.
- [x] The selection does not update when I click on a card with the split screen already opened. Also when I reload the page, there is no card selected when the split screen is open.
- [x] When I have a lot of lists next to each other (e.g. 10), the split screen is not visible, although it is open and there are two scrollbars at the bottom.

**Out of scope  (tracked separately)**
* Sometimes opening the split screen, caused the save icon to appear. However I could not reliably reproduce this one.
* In a version board: When I press "Add new list" before all lists are loaded, the select shows no versions. When I close the modal, and press "Add" again, the select isn't loaded at all. There is also an error.

### CODE REVIEW

**Comment by oliverguenther** [comment]
Screenshot from 2020-03-30 14-40-34

@HDinger there is a minor style issue with the back button, we could force it to `align-self: flex-start` but maybe you have a better idea?

**Comment by oliverguenther** [comment]
Ticket: https://community.openproject.com/projects/openproject/work_packages/30918

File: frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts [hunk 1]
```diff
@@ -11,27 +11,18 @@ import {WorkPackageViewFiltersService} from "core-app/modules/work_packages/rout
...
 import {UrlParamsHelperService} from "core-components/wp-query/url-params-helper";
 import {StateService} from "@uirouter/core";
-import {DebouncedEventEmitter} from "core-components/angular/debounced-event-emitter";
-import {skip} from "rxjs/internal/operators";
-import {ApiV3Filter} from "core-components/api/api-v3/api-v3-filter-builder";
+import {debounceTime, skip, take} from "rxjs/internal/operators";
 import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
-import {componentDestroyed} from "@w11k/ngx-componentdestroyed";
+import {Observable} from "rxjs";
+import {BoardFiltersService} from "core-app/modules/boards/board/board-filter/board-filters.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/9fa518c8247ead78913e82abf903114474bb7670/frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts#L18"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/boards/boards-root/boards-root.component.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import {QueryUpdatedService} from "core-app/modules/boards/board/query-updated/q
```
Comment (openprojectci) [inline frontend/src/app/modules/boards/boards-root/boards-root.component.ts]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/9fa518c8247ead78913e82abf903114474bb7670/frontend/src/app/modules/boards/boards-root/boards-root.component.ts#L7"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/9fa518c8247ead78913e82abf903114474bb7670/frontend/src/app/modules/boards/board/add-list-modal/add-list-modal.component.ts#L44">frontend/src/app/modules/boards/board/add-list-modal/add-list-modal.component.ts#L44</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/9fa518c8247ead78913e82abf903114474bb7670/frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts#L54">frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts#L54</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/9fa518c8247ead78913e82abf903114474bb7670/frontend/src/app/modules/boards/new-board-modal/new-board-modal.component.ts#L41">frontend/src/app/modules/boards/new-board-modal/new-board-modal.component.ts#L41</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by lindenthal** [comment]
Hi @oliverguenther,
I just had a quick look at PP.

http://openproject-pr-8230-make-boards-a-3338efd8.my.pullpreview.com/projects/your-scrum-project/boards/3/details/28/overview

After activating the split view I can not switch to another work package. Can you please have a look. Am I using an outdated version?

**Comment by lindenthal** [comment]
@oliverguenther: did you consider to add an info icon to the toolbar, similar to the work packages view?

**Comment by HDinger** [comment]
> @oliverguenther: did you consider to add an info icon to the toolbar, similar to the work packages view?

When we add the info icon to the toolbar, we should briefly think about what this means for the selection. In the board, I can easily select multiple cards without any keyboard help (one card per list). The question I then ask myself is which card will appear when I press the info icon. The selected one of the first list? Or the one I selected last (which is currently not possible)? Since we have not decided this yet, it was easier to show only the icon in the card for now.

File: frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts (hunk 1 above)
Comment (openprojectci) [inline frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts:18]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/bb255f1686bcbf57416091d8e31a1e3ba9bf1ea4/frontend/src/app/modules/boards/board/board-filter/board-filter.component.ts#L18"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/boards/boards-root/boards-root.component.ts (hunk 2 above)
Comment (openprojectci) [inline frontend/src/app/modules/boards/boards-root/boards-root.component.ts:7]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/8d0ecb4b3ed813f91bb58c934f28a2845d090292/frontend/src/app/modules/boards/boards-root/boards-root.component.ts#L7"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/bb255f1686bcbf57416091d8e31a1e3ba9bf1ea4/frontend/src/app/components/modals/wp-destroy-modal/wp-destroy.modal.ts#L42">frontend/src/app/components/modals/wp-destroy-modal/wp-destroy.modal.ts#L42</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/bb255f1686bcbf57416091d8e31a1e3ba9bf1ea4/frontend/src/app/modules/boards/board/add-list-modal/add-list-modal.component.ts#L44">frontend/src/app/modules/boards/board/add-list-modal/add-list-modal.component.ts#L44</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/bb255f1686bcbf57416091d8e31a1e3ba9bf1ea4/frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts#L54">frontend/src/app/modules/boards/board/inline-add/board-inline-add-autocompleter.component.ts#L54</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/bb255f1686bcbf57416091d8e31a1e3ba9bf1ea4/frontend/src/app/modules/boards/new-board-modal/new-board-modal.component.ts#L41">frontend/src/app/modules/boards/new-board-modal/new-board-modal.component.ts#L41</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by oliverguenther** [comment]
@HDinger the failing PR state is due to PullPreview having failed to start. I stopped the auto-build but the status is still "failed" now.

**Review by HDinger — CHANGES_REQUESTED** [review]
I think there is one point we did not think about in the beginning: When I am on an action board (e.g. status) and I change the status in the split screen, the list should to be updated, right? 

Other than that. I also found a few other issues:

* When being on a free board, and I update the title, type or assignee in the split screen, the belonging card does not change.

* Sometimes opening the split screen, caused the save icon to appear. However I could not reliably reproduce this one.

* When I have the split screen open, a click on the info icon of another card only selects this card. I would expect it, to show the details in the split screen.

* The selection does not update when I click on a card with the split screen already opened. Also when I reload the page, there is no card selected when the split screen is open.

* When I have a lot of lists next to each other (e.g. 10), the split screen is not visible, although it is open and there are two scrollbars at the bottom.

* In a version board: When I press "Add new list" before all lists are loaded, the select shows no versions. When I close the modal, and press "Add" again, the select isn't loaded at all. There is also an error:
Bildschirmfoto 2020-04-02 um 15 09 01



Some styling issues:

* I think we should switch the info and the close icon on the card. Close should always be the last one.

* There is no space between the filter section and the right screen border.

* There are some alignment issues: 
Bildschirmfoto 2020-04-02 um 14 44 53

**Comment by oliverguenther** [comment]
> The selection does not update when I click on a card with the split screen already opened. Also when I reload the page, there is no card selected when the split screen is open.

Selections in the lists were deliberately disabled. We cannot have both the selection service from within the query space AND the focus service from outside. This will not work

**Comment by lindenthal** [comment]
Hi @oliverguenther,

I just had a quick look at this change. Looks great!

I have one minor remark regarding the position of the info icon the card. Currently it is on the very right. I suggest to keep the remove icon on the very right. What do you think? grafik

**Comment by lindenthal** [comment]
Hi @oliverguenther,
I am not sure if I did understand you 100%. You said it is technically not possible to activate the split view from the toolbar similiar to the wp list view, right?

Best
Niels

File: frontend/src/app/modules/boards/boards-root/boards-root.component.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import {BoardAssigneeActionService} from "core-app/modules/boards/board/board-ac
```
Comment (openprojectci) [inline frontend/src/app/modules/boards/boards-root/boards-root.component.ts:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/bb255f1686bcbf57416091d8e31a1e3ba9bf1ea4/frontend/src/app/modules/boards/boards-root/boards-root.component.ts#L8"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts [hunk 4]
```diff
@@ -43,6 +43,7 @@ import {IFieldSchema} from "core-app/modules/fields/field.base";
 import {EditFieldHandler} from "core-app/modules/fields/edit/editing-portal/edit-field-handler";
 import {EditingPortalService} from "core-app/modules/fields/edit/editing-portal/editing-portal-service";
 import {EditFormRoutingService} from "core-app/modules/fields/edit/edit-form/edit-form-routing.service";
+import {ResourceChangesetCommit} from "core-app/modules/fields/edit/services/hal-resource-editing.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts:47]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/bb255f1686bcbf57416091d8e31a1e3ba9bf1ea4/frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L47"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Review by HDinger — APPROVED** [review]

**Pull request merged**